### PR TITLE
Added name records pagination Query Flow

### PR DESCRIPTION
**Pull request by DiamondJaws**
Changelog:
-Paginate name_records (MAX 30)
-tests moved in ./tests/
-added tests for name_records query

### CODE REVIEW

File: src/query/query_utils.rs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+use crate::{
+    error::ContractError,
+    models::PublicNameRecord,
+    state::{NAME_METADATA, NAME_RECORDS},
+};
+use super::ReadonlyContext;
+
+pub fn build_public_name_record(ctx: &ReadonlyContext, cannonical_name: String) -> Result<PublicNameRecord, ContractError> {
```
Comment (dgabriele) [inline src/query/query_utils.rs]: Could possibly make this into a method on the NameRecord struct

File: src/msg.rs [hunk 2]
```diff
@@ -43,6 +43,15 @@ pub struct RenderQueryMsg {
     pub context: Option<Value>,
 }
 
+#[cw_serde]
+pub struct NameRecordsQueryMsg {
+    pub limit: u8,
+    pub cursor: Option<String>,
+    pub network_den: Option<String>
```
Comment (dgabriele) [inline src/msg.rs]: Often times, what you call `network_den` here is referred to as `prefix` in cosmwasm contexts. Could possibly rename it to that (or more fully `network_prefix`)

**Review by dgabriele — CHANGES_REQUESTED** [review]
🔥 I like the independent test module. And now that you have implemented this pattern for pagination, you can pretty much reuse the same logic anywhere else now. Quite useful, if I may say so :)

Just left a couple naming-related comments. Could you run `cargo fmt` when you're ready to merge?

File: src/query/query_utils.rs (hunk 1 above)
Comment (DiamondJaws) [reply src/query/query_utils.rs]: done

File: src/msg.rs (hunk 2 above)
Comment (DiamondJaws) [reply src/msg.rs]: done

**Pull request merged**